docs/proposals/etcd-druid.md

```diff
@@ -0,0 +1,24 @@
+# Integrating etcd-druid with Gardener.
+
+Etcd is currently deployed by garden-controller-manager as a Statefulset. The sidecar container spec contains details pertaining to cloud-provider object-store which is injected into the statefulset via a mutable webhook running as part of the gardener extension [story](https://github.com/gardener/gardener/blob/master/docs/extensions/controlplane-webhooks.md#what-needs-to-be-implemented-to-support-a-new-cloud-provider).  This approach restricts the operations on etcd such as scale-up and upgrade. Etcd-druid will eliminate the need to hijack statefulset creation to add cloudprovider details. It has been designed to provide an intricate control over the procedure of deploying and maintaining etcd. The roadmap for etcd-druid can be found [here](https://github.com/gardener/etcd-druid/issues/2). etcd-
+
+This document explains how Gardener deploys etcd and what resources it creates for etcd-druid to deploy an etcd cluster.
+
+## Resources required by etcd-druid (created by Gardener)
+
+* Secret containing credentials to access backup bucket in Cloud provider object store.
+* TLS server and client secrets for etcd and backup-sidecar
+* Etcd CRD resource that contains parameters pertaining to etcd, backup-sidecar and cloud-provider object store.
+
+When an etcd resource is created in the cluster, the druid acts on it by creating an etcd statefulset, a service and a configmap containing etcd bootstrap script. The secrets containing the infrastructure credentials and the TLS certificates are mounted as volumes. 
```

Comment:
And if no secret/information regarding the backup bucket is stated, does the etcd-druid then only create an etcd `StatefulSet` together with the `backup-restore` sidecar that is only used for data validation, corruption checks, etc., but not for backup?

Reply:
The etcd-backup-restore sidecar supports it, so druid will be able to do that

Reply:
Good, thanks! Can you please add this description to the paragraph here?